metals/src/main/scala/scala/meta/internal/builds/GradleBuildTool.scala

```diff
@@ -7,20 +7,59 @@ import java.nio.file.Path
 import scala.util.Properties
 
 import scala.meta.internal.metals.BuildInfo
+import scala.meta.internal.metals.MetalsEnrichments._
 import scala.meta.internal.metals.UserConfiguration
 import scala.meta.io.AbsolutePath
 
+import coursierapi.Credentials
+import coursierapi.IvyRepository
+import coursierapi.MavenRepository
+import coursierapi.Repository
+
 case class GradleBuildTool(userConfig: () => UserConfiguration)
     extends BuildTool
     with BloopInstallProvider {
 
   private val initScriptName = "init-script.gradle"
   private val gradleBloopVersion = BuildInfo.gradleBloopVersion
-  private val initScript =
+  private def initScript = {
+    def authString(cr: Credentials) =
+      if (cr == null) ""
+      else
+        s"""|
+            |      credentials {
+            |        username ${cr.getUser()}
+            |        password ${cr.getPassword()}
+            |      }
+            |""".stripMargin
+    val repos = Repository.defaults.asScala.collect {
+      case mr: MavenRepository if mr != Repository.central =>
+        // filter central etc.
+        s"""|    maven {
+            |      url "${mr.getBase()}" ${authString(mr.getCredentials())}
+            |    }""".stripMargin
+
+      case ir: IvyRepository =>
```

Comment:
Would you be able to add some test for transforming repos to the Gradle style? This can be unit tests or tests within GradleLspSuite. The latter might be harder to get going. Unit tests might be enough, though you might need to maybe extract `repos` to a testable method such as inside GradleBuildTool companion object:
```
def transformRepos(repos: List[Repository] = Repository.defaults) = ???
```
what do you think?

Reply:
I added some unit tests. :)